File: scripts/generate_products_sql.py
import json
import re

def parse_price(price_str):
    # Extract all price matches
    # Input format examples: 
    # "EGP47,280.00 \n..."
    # "EGP34,000.00"
    matches = re.findall(r'EGP\s*([\d,]+\.?\d*)', price_str)
    if matches:
        # Return the last one (usually the discounted/current price)
        # Remove commas
        try:
            return float(matches[-1].replace(',', ''))
        except ValueError:
            return 0.0
    return 0.0

def parse_attributes(title):
    title_lower = title.lower()
    
    # Horsepower
    hp = "1.5" # Default
    # Look for 1.5, 2.25, 3, 4, 5, 6, 7.5
    hp_match = re.search(r'(\d+(?:\.\d+)?)\s*(?:hp|حصان)', title_lower)
    if hp_match:
        hp = hp_match.group(1)
    
    # Type
    p_type = "wall" # Default (Split)
    if "stand" in title_lower or "ستند" in title_lower:
        p_type = "freestand"
    elif "ceiling" in title_lower or "سقفى" in title_lower or "floor" in title_lower or "أرضى" in title_lower:
        p_type = "floor_ceiling"
    elif "concealed" in title_lower or "كونسيلد" in title_lower:
        p_type = "concealed"
    
    # Cooling Type
    cooling = "cool_only"
    if "heat" in title_lower or "ساخن" in title_lower:
        cooling = "cool_heat"
        
    return hp, p_type, cooling

def generate_sql():
    try:
        with open('scraped_products.json', 'r', encoding='utf-8') as f:
            products = json.load(f)
    except FileNotFoundError:
        print("Error: scraped_products.json not found.")
        return

    sql_statements = []
    
    # 1. Ensure Brands Exist
    sql_statements.append("-- Ensure Brands Exist (Carrier, Midea)")
    sql_statements.append("INSERT INTO brands (name, name_ar) VALUES ('Carrier', 'كاريير') ON CONFLICT (name) DO NOTHING;")
    sql_statements.append("INSERT INTO brands (name, name_ar) VALUES ('Midea', 'ميديا') ON CONFLICT (name) DO NOTHING;")
    sql_statements.append("")

    sql_statements.append("-- Insert Products")
    
    for p in products:
        title = p.get('title', '').replace("'", "''") # Escape single quotes
        brand_name = p.get('brand', 'Carrier')
        price_str = p.get('price', '')
        image_url = p.get('image_url', '')
        product_link = p.get('link', '')
        
        price = parse_price(price_str)
        hp, p_type, cooling = parse_attributes(title)
        
        # Determine features array (optional, but good for backup)
        features_json = json.dumps([hp + " HP", p_type, cooling], ensure_ascii=False)
        
        # Prepare INSERT statement
        # We assume columns: name, price, brand_id, image_url, is_active, type, horsepower, cooling_type, description
        # Using ON CONFLICT (name) to update if exists
        
        # Note: We need to handle the subquery for brand_id
        
        sql = f"""
INSERT INTO products (
    name, 
    price, 
    brand_id, 
    image_url, 
    type, 
    horsepower, 
    cooling_type, 
    description, 
    is_active,
    stock
) VALUES (
    '{title}', 
    {price}, 
    (SELECT id FROM brands WHERE name = '{brand_name}' LIMIT 1), 
    '{image_url}', 
    '{p_type}', 
    '{hp}', 
    '{cooling}', 
    'Scraped from ModernCool: {product_link}', 
    true,
    10
)
ON CONFLICT (name) DO UPDATE SET 
    price = EXCLUDED.price,
    image_url = EXCLUDED.image_url,
    type = EXCLUDED.type,
    horsepower = EXCLUDED.horsepower,
    cooling_type = EXCLUDED.cooling_type,
    updated_at = NOW();
"""
        sql_statements.append(sql.strip())

    # Write to file
    with open('supabase_seed_products.sql', 'w', encoding='utf-8') as f:
        f.write('\n\n'.join(sql_statements))
    
    print(f"Generated {len(products)} product insert statements in supabase_seed_products.sql")

if __name__ == "__main__":
    generate_sql()

File: scripts/generate_products_sql.js
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

function parsePrice(priceStr) {
    if (!priceStr) return 0;
    // Match "EGP" followed by numbers/commas
    const matches = priceStr.match(/EGP\s*([\d,]+\.?\d*)/g);
    if (matches && matches.length > 0) {
        // Use the last match (often "Current Price")
        const last = matches[matches.length - 1];
        const numStr = last.replace(/EGP\s*/, '').replace(/,/g, '');
        return parseFloat(numStr) || 0;
    }
    // Fallback if no EGP prefix but assumes it's a number string
    return 0;
}

function parseAttributes(title) {
    if (!title) return { hp: "1.5", type: "wall", cooling: "cool_only" };
    const titleLower = title.toLowerCase();

    // HP
    let hp = "1.5";
    const hpMatch = titleLower.match(/(\d+(?:\.\d+)?)\s*(?:hp|حصان)/);
    if (hpMatch) hp = hpMatch[1];

    // Type
    let type = "wall";
    if (titleLower.includes("stand") || titleLower.includes("ستند")) type = "freestand";
    else if (titleLower.includes("ceiling") || titleLower.includes("سقفى") || titleLower.includes("floor") || titleLower.includes("أرضى")) type = "floor_ceiling";
    else if (titleLower.includes("concealed") || titleLower.includes("كونسيلد")) type = "concealed";

    // Cooling
    let cooling = "cool_only";
    if (titleLower.includes("heat") || titleLower.includes("ساخن")) cooling = "cool_heat";

    return { hp, type, cooling };
}

try {
    const jsonPath = path.join(__dirname, '../scraped_products.json');
    const outPath = path.join(__dirname, '../supabase_seed_products.sql');

    console.log(`Reading from ${jsonPath}`);
    if (!fs.existsSync(jsonPath)) {
        console.error("JSON file not found!");
        process.exit(1);
    }

    const data = fs.readFileSync(jsonPath, 'utf8');
    const products = JSON.parse(data);

    let sql = "-- Ensure Brands Exist\n";
    sql += "INSERT INTO brands (name, name_ar) VALUES ('Carrier', 'كاريير') ON CONFLICT (name) DO NOTHING;\n";
    sql += "INSERT INTO brands (name, name_ar) VALUES ('Midea', 'ميديا') ON CONFLICT (name) DO NOTHING;\n\n";
    sql += "-- Insert Products\n";

    let count = 0;
    products.forEach(p => {
        const title = (p.title || '').replace(/'/g, "''");
        const price = parsePrice(p.price || '');
        const { hp, type, cooling } = parseAttributes(p.title || '');
        const brand = p.brand || 'Carrier';

        // Skip if almost empty
        if (!title) return;

        sql += `INSERT INTO products (name, price, brand_id, image_url, type, horsepower, cooling_type, description, is_active, stock) VALUES (
    '${title}', 
    ${price}, 
    (SELECT id FROM brands WHERE name = '${brand}' LIMIT 1), 
    '${p.image_url}', 
    '${type}', 
    '${hp}', 
    '${cooling}', 
    'Scraped from ModernCool: ${p.link}', 
    true,
    10
) ON CONFLICT ON CONSTRAINT products_name_unique DO UPDATE SET 
    price = EXCLUDED.price,
    image_url = EXCLUDED.image_url,
    type = EXCLUDED.type,
    horsepower = EXCLUDED.horsepower,
    cooling_type = EXCLUDED.cooling_type,
    updated_at = NOW();\n\n`;
        count++;
    });

    fs.writeFileSync(outPath, sql);
    console.log(`Successfully generated SQL for ${count} products at ${outPath}`);

} catch (e) {
    console.error("Error generating SQL:", e);
    process.exit(1);
}
